### PR TITLE
PanZoom: Expose Zoom and Pan Api

### LINKED ISSUE #3141: set zoom in zoom out using external controls
Hi,

I have a requirement where I need to zoom in and zoom out over a Quantitative time scale using some external buttons (say one for zooming in and another for zooming out).

Is there a way I can get and set zoom levels for  a chart as per the click on these buttons.

### PR DESCRIPTION
**Pull request by CalvinFernandez**
Expose pan and zoom api to allow users to imperatively pan and
zoom the chart independently of the mousewheel or pinch events.

closes #3141

### CODE REVIEW

File: src/interactions/panZoomInteraction.ts
Comment (themadcreator) [inline src/interactions/panZoomInteraction.ts:77]: I think this is going to conflict with the changes I made to enable pan/zoom on category scales. So, can we wait until that other is merged? I can pull this down and update the PR once it's merged.

**Review by themadcreator — CHANGES_REQUESTED** [review]
Conflicts incoming

File: src/interactions/panZoomInteraction.ts
Comment (CalvinFernandez) [reply src/interactions/panZoomInteraction.ts:77]: yep noticed major conflicts -- I'm totally fine waiting

**Review by themadcreator — APPROVED** [review]

**Review by hellochar — APPROVED** [review]

**Comment by yocontra** [comment]
Are there any docs live for this yet? Couldn't find any.

**Comment by CalvinFernandez** [comment]
@contra working on this now sorry for the delay

**Comment by yocontra** [comment]
@CalvinFernandez No prob, just wondering if I was missing something - I thought the docs were autogenerated from the source

**Pull request merged**
